From b56225808d08b772f12a6e6cb6e010b12abe56cb Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@spacemit.com>
Date: Tue, 16 Apr 2024 16:26:12 +0800
Subject: [PATCH 0597/1202] clear compile warning:

drivers/gpu/drm/spacemit/lt8911exb.c: In function 'lt8911exb_video_check':
drivers/gpu/drm/spacemit/lt8911exb.c:764:22: warning: unused variable 'temp' [-Wunused-variable]
  764 |         unsigned int temp;
      |                      ^~~~
drivers/gpu/drm/spacemit/lt8911exb.c: In function 'lt8911exb_probe':
drivers/gpu/drm/spacemit/lt8911exb.c:1363:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
 1363 |         struct mipi_dsi_device_info info = {
      |         ^~~~~~

Change-Id: If7b1a0eed35e9655b96cf89d3c6df71bae2ceffe
---
 drivers/gpu/drm/spacemit/lt8911exb.c     | 15 ++-------------
 drivers/gpu/drm/spacemit/spacemit_hdmi.c |  1 -
 2 files changed, 2 insertions(+), 14 deletions(-)

diff --git a/drivers/gpu/drm/spacemit/lt8911exb.c b/drivers/gpu/drm/spacemit/lt8911exb.c
index 2fa4d54bf450..8e725e197fef 100644
--- a/drivers/gpu/drm/spacemit/lt8911exb.c
+++ b/drivers/gpu/drm/spacemit/lt8911exb.c
@@ -208,15 +208,6 @@ static const struct regmap_config lt8911exb_regmap_config = {
 	.max_register = 0xff,
 };
 
-static struct lt8911exb *bridge_to_lt8911exb(struct drm_bridge *bridge)
-{
-	return container_of(bridge, struct lt8911exb, bridge);
-}
-
-static struct lt8911exb *connector_to_lt8911exb(struct drm_connector *connector)
-{
-	return container_of(connector, struct lt8911exb, connector);
-}
 
 static struct lt8911exb *panel_to_lt8911exb(struct drm_panel *panel)
 {
@@ -761,7 +752,6 @@ void lt8911exb_setup(struct lt8911exb *lt8911exb)
 
 void lt8911exb_video_check(struct lt8911exb *lt8911exb)
 {
-	unsigned int temp;
 	unsigned int reg;
 	unsigned int temp2;
 
@@ -1358,14 +1348,14 @@ static int lt8911exb_probe(struct i2c_client *client,
 	char lcd_path[60];
 	const char *lcd_name;
 
-	DRM_INFO("%s()\n", __func__);
-
 	struct mipi_dsi_device_info info = {
 		.type = IT8911_DSI_DRIVER_NAME,
 		.channel = 0, //0,
 		.node = NULL,
 	};
 
+	DRM_INFO("%s()\n", __func__);
+
 	if (!i2c_check_functionality(client->adapter, I2C_FUNC_I2C)) {
 		dev_err(&client->dev, "Failed check I2C functionality");
 		return -ENODEV;
@@ -1553,7 +1543,6 @@ static int lt8911exb_dsi_probe(struct mipi_dsi_device *dsi)
 	if (ret < 0) {
 		dev_err(&dsi->dev, "failed to attach dsi to host\n");
 		mipi_dsi_device_unregister(dsi);
-		return ERR_PTR(ret);
 	}
 
 	return ret;
diff --git a/drivers/gpu/drm/spacemit/spacemit_hdmi.c b/drivers/gpu/drm/spacemit/spacemit_hdmi.c
index c0aa17eec84e..84f7a7e2e591 100644
--- a/drivers/gpu/drm/spacemit/spacemit_hdmi.c
+++ b/drivers/gpu/drm/spacemit/spacemit_hdmi.c
@@ -549,7 +549,6 @@ void hdmi_init (struct spacemit_hdmi *hdmi, int pixel_clock, int bit_depth){
 static int spacemit_hdmi_setup(struct spacemit_hdmi *hdmi,
 			   struct drm_display_mode *mode)
 {
-	u32 value;
 	struct hdmi_data_info *hdmi_data = hdmi->hdmi_data;
 	int bit_depth = TEN_BPP;
 
-- 
2.47.0

